Navigation Menu: render item without link

retrofox opened the pull request:
It handles the case when the item has not defined a link, rendering the item with a `<span>` element instead of an anchor `<a>`.

## Description
<!-- Please describe what you have changed or added -->

## How has this been tested?
Add an item without a link from the Editor canvas.
Confirm the markup sets an <a> anchor element <a> to those items that have defined a link, and sets a <span> element when there aren't. Pay attention to the CSS class as well.

<img width="804" alt="Screen Shot 2019-11-13 at 12 02 06 PM" src="https://user-images.githubusercontent.com/77539/68775620-91485680-060d-11ea-8535-a887c9819104.png">

## Screenshots <!-- if applicable -->

![image](https://user-images.githubusercontent.com/77539/68775402-2dbe2900-060d-11ea-8b37-b0ea67aa6606.png)

## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [ ] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->.

mtias committed:
Rename "Navigation Menu Item" to "Link".

retrofox committed:
navigation-menu: render item withour link
It handles the case when the item has not defined a linl, rendering the item with an <span> element instrad of an anchor <a>.

obenland reviewed (changes requested):
All values being output into an element attribute need to be run through `esc_attr()`.

tellthemachines commented on packages/block-library/src/navigation-menu/index.php:
```diff
@@ -97,31 +97,36 @@ function build_navigation_menu_html( $block, $colors ) {
+				'</a>';
+		} else {
+			$item_content_markup =
+				'<label
```
Any reason for this to be a `<label>`? Labels with no form fields associated have no semantic value, so you might as well use a `<span>` and avoid having to override label-specific styles. I'm also thinking this could become weirdly unsemantic if we ever decide to add other types of markup inside the nav items.

draganescu replied:
+1 - a span is better than an empty label

retrofox replied:
👍 I changed for a `<label>` in the last commit. Let me rever it.

tellthemachines commented on packages/block-library/src/navigation-menu-item/editor.scss:
```diff
@@ -30,6 +30,10 @@
 		display: block;
 	}
+
+	.wp-block-navigation-menu-item__label {
```
I don't believe this classname is ever applied on the editor side, is it?

tellthemachines reviewed (changes requested):
Thanks for working on this! Just a few fairly minor comments/questions below.

retrofox committed:
navigation-menu: fix unclosing quote

retrofox committed:
navigation-menu: escape all html attrs

retrofox committed:
navigation-menu: fix lint-php errors

draganescu commented on packages/block-library/src/navigation-menu/index.php:
```diff
@@ -97,31 +97,36 @@ function build_navigation_menu_html( $block, $colors ) {
+				'<a
+					class="wp-block-navigation-menu-item__link ' . $colors['text_css_classes'] . '"
+					' . $colors['text_inline_styles'] .
+					' href="' . $menu_item['attrs']['url'] .
```
@retrofox I think we could do better with this code as the concatenation combined with HTML and ternary results in a soup of chars :) very hard to debug/ read.

retrofox replied:
Fair enough. I changed the code before reading this comment, so let me know if it's good to go or I should keep improving the code.

```php
'<a
	class="wp-block-navigation-menu-item__link ' . $colors['text_css_classes'] . '"
	' . $colors['text_inline_styles'] .
	' href="' . $url . '"' .
	( $title ? ' title="' . $title . '"' : '' ) .
	( $opens_in_new_tab ? ' target="_blank"' : '' ) .
'>' .
	( $label ? $label : '' ) .
'</a>';
```

retrofox committed:
navigation-menu: remove unused CSS. Rename class

retrofox committed:
navigation-menu: simplify making the markup

retrofox committed:
navigation-menu: set tezxt item styles

obenland committed:
Updating a few more menu item mentions

If we end up going with a different term, at least we'll have a better idea what it all is that needs updating.

draganescu commented on packages/block-library/src/navigation-menu/index.php:
```diff
@@ -75,32 +75,34 @@ function render_block_navigation_menu( $attributes, $content, $block ) {
+					class="wp-block-navigation-menu-item__text ' . $colors['text_css_classes'] . '"
+					' . $colors['text_inline_styles'] .
+				'>' .
+					( isset( $menu_item['attrs']['label'] ) ? $label : '' ) .
```
This is the same `$label` as above right?

retrofox replied:
yes :-( Going to update

retrofox committed:
navigation-menu: simply setting item label

retrofox committed:
e2e-tests: fix navigation-link test

jasmussen commented:
@retrofox is this one outdated now that we have a placeholder state for empty menu items?

![nav](https://user-images.githubusercontent.com/1204802/111609010-24813580-87da-11eb-93cb-e74701291307.gif)

retrofox commented:
> @retrofox is this one outdated now that we have a placeholder state for empty menu items?

Yeah, pretty sure it is! let me close it. Thanks for the ping 👍
